substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/phases/IntrinsifyMethodHandlesInvocationPlugin.java

```diff
@@ -516,13 +528,26 @@ private void processInvokeWithMethodHandle(GraphBuilderContext b, Replacements r
 
             JavaKind returnResultKind = b.getInvokeReturnType().getJavaKind().getStackKind();
             ValueNode transplantedSingleFunctionality = null;
+            ValueNode transplantedNewInstance = null;
             if (singleFunctionality instanceof Invoke) {
+
                 Invoke singleInvoke = (Invoke) singleFunctionality;
                 MethodCallTargetNode singleCallTarget = (MethodCallTargetNode) singleInvoke.callTarget();
                 ResolvedJavaMethod resolvedTarget = lookup(singleCallTarget.targetMethod());
 
-                maybeEmitClassInitialization(b, singleCallTarget.invokeKind() == InvokeKind.Static, resolvedTarget.getDeclaringClass());
+                if (singleNewInstance != null) {
+                    ResolvedJavaType type = null;
+                    if (singleNewInstance instanceof DynamicNewInstanceNode) {
+                        type = lookup(originalProviders.getConstantReflection().asJavaType(((DynamicNewInstanceNode) singleNewInstance).getInstanceType().asConstant()));
+                    }
+                    if (singleNewInstance instanceof NewInstanceNode) {
+                        type = lookup(((NewInstanceNode) singleNewInstance).instanceClass());
+                    }
+                    ValueNode newInstance = b.add(new NewInstanceNode(type, true));
```

Comment:
You also need a `maybeEmitClassInitialization` before the `NewInstanceNode`, otherwise you might allocate a non-initialized class.